### DISCUSSION TITLE
KAM perimeter

### DISCUSSION THREAD

**Discussion by Fjvs1** [mtex-toolbox/mtex#1144] [Ask Anything] [answered]
Hi there !

I'm trying to find results from another guy who used OIM for his KAM analysis.

It seams he used perimeter's pixels for his KAM and not every pixels in the neighborhood.
![image](https://user-images.githubusercontent.com/87763359/126489729-14f74abf-d7b3-40e7-9a15-c21cd0c9cdf8.png)

Is it possible to do the same with MTEX? 
I tried to soustract a KAM with order 2 from a KAM order 3 to have a perimetered KAM order 3, but it's not as easy as that.

Thanks a lot for your answers.
FJVS1.

**Comment by ralfHielscher** [comment, marked as answer]
Have a look at [EBSDHex/KAM](https://github.com/mtex-toolbox/mtex/blob/develop/EBSDAnalysis/%40EBSDhex/KAM.m)

There you have a loop of the different radii. You might want to omit this loop at take only the last radius.

Ralf.

**Reply by Fjvs1** [reply]
Thanks Ralf.

I omitted a detail, I use square grid. The illustration from OIM is in hexagonal grid (my bad). 
I checked your script EBSDSquare/KAM. 
Do I have only _"  for i = -order:order;  for j = -order:order"_ (lines 88 89) to change ?

Great job for the scripts :)
